src/main/resources/com/nec/cyclone/cpp/cyclone/cyclone_utils.hpp

```diff
@@ -28,38 +28,60 @@
 #include "frovedis/dataframe/join.hpp"
 
 namespace cyclone {
+  template<typename T>
+  void print_vec(char *name, std::vector<T> a) {
```

Comment:
I think we already define `auto& operator<<` for `std::vector` and `std::tuple` below.  Perhaps this definition can be further shortened to make use of that, since it only a name in front.

Reply:
I've just moved this around. When using it for debugging within this file, it isn't defined if it is at the end of the namespace definition. 

You're right that we could probably make use of the additionally defined operators, but I think that should be a different PR.